azuki-script-generation/src/main/kotlin/com/anaplan/engineering/azuki/script/generation/ScriptGenerationAction.kt

```diff
@@ -3,16 +3,29 @@ package com.anaplan.engineering.azuki.script.generation
 import com.anaplan.engineering.azuki.core.system.Action
 import com.anaplan.engineering.azuki.core.system.ParallelAction
 
-interface ScriptGenerationAction : Action {
+interface ScriptGenerationAction<in E : ScriptGenerationEnvironment> : Action {
+
+    fun getActionScript(environment: E): String
+}
+
+/**
+ * Script generation action that doesn't use its environment.
+ */
+interface IgnoreEnvScriptGenerationAction<in E : ScriptGenerationEnvironment> : ScriptGenerationAction<E> {
+
     fun getActionScript(): String
+
+    override fun getActionScript(environment: E) = getActionScript()
 }
 
-class ScriptGenerationParallelAction(actions: List<List<ScriptGenerationAction>>) :
-    ParallelAction<ScriptGenerationAction>(actions), ScriptGenerationAction {
+typealias NoEnvScriptGenerationAction = IgnoreEnvScriptGenerationAction<NoScriptGenerationEnvironment>
```

Comment:
❓ Why bother with the type alias? Couldn't this have worked as
```suggestion
interface NoEnvScriptGenerationAction : ScriptGenerationAction<NoScriptGenerationEnvironment> {

    fun getActionScript(): String
    override fun getActionScript(environment: E) = getActionScript()
}
```

Reply:
That's fused the type parameter to `NoScriptGenerationEnvironment`, and there are real-world situations where we want to have an environment for some actions but not others, which would mean that the type parameter going into `ScriptGenerator` couldn't be `NoScriptGenerationEnvironment` and consequently all of the code in the script generator would be expecting descendants of `ScriptGenerationAction<E>` for some `E != NoScriptGenerationEnvironment`.